lib/fs.js

```diff
@@ -1160,7 +1160,7 @@ function FSWatcher() {
   this._handle.onchange = function(status, event, filename) {
     if (status < 0) {
       self._handle.close();
-      self.emit('error', errnoException(status, 'watch'));
+      self.emit('error', errnoException(status, 'watch ' + filename));
```

Comment:
I meant do something like:

```
var err = errnoException(status, 'watch');
err.filename = filename;
```

I personally find it more useful to have the filename separate. Maybe others feel differently. Maybe do both.